src/driver/swim.c

```diff
@@ -318,26 +317,34 @@ int swim_entry() {
   // 128 cycles (@8MHz) i.e. 16us. We currently just use it as a sanity
   // check to make sure the (right) device responds.
   timeout = MICROS_TO_CYCLES(30) / 6;
-  while (!READ_PIN() && --timeout)
+  while (!READ_PIN(SWIM) && --timeout)
     ;  // ~6 cycles
   int duration = get_ccount() - counter;
   if (!timeout) return SWIM_ERROR_SYNC_TIMEOUT_2;
 
   esp8266_leave_critical(state);
-  sync_ccount(counter + duration +
-              24);  // Need to wait at least 300ns (=24 cycles)
+
+  SET_PIN_HIGH(SWIM);
+  PIN_AS_INPUT(SWIM);
+  sync_ccount(counter + duration + 24 );
+
+#if(FIRMWARE_VERSION_MAJOR + FIRMWARE_VERSION_MINOR == 0)
```

Comment:
I’m not sure I understand your rationale here. This seems to always evaluate to false?

Reply:
The idea of FIRMWARE_VERSION is to signal it to the client, so that it can know that certain features are supported.

But I think I understand what you’re trying to do. In order for it to work, you’ll need to reset FIRMWARE_VERSION back to 0 in version.h. Then one can flash this version and have the implicit reset support. That also means that the #if is not needed. 

Then in another commit, once the respective feature in stm8flash is implemented (tracking in issue #7), we can increase to 0.1 again and delete the line.

So please replace the #f with a // TODO marker referencing https://github.com/rumpeltux/esp-stlink/issues/7 and change version.h back to 0.